src/components/Widgets/TaskPropertiesWidget/TaskPropertiesWidget.jsx

```diff
@@ -19,45 +19,91 @@ const descriptor = {
-                </div>
-              )
-            })
-          }
-        </div>
-      )
-    })
+const TaskPropertiesWidget = (props) => {
+  const taskList = _get(props.taskBundle, 'tasks') || [props.task];
+  const initialCollapsedState = {};
+
+  taskList.forEach((task) => {
+    const featurePropertiesList = AsMappableTask(task).osmFeatureProperties(props.osmElements);
+    featurePropertiesList.forEach((_, index) => {
+      initialCollapsedState[`${task.id}-${index}`] = true;
+    });
+  });
+
+  const [collapsed, setCollapsed] = useState(initialCollapsedState); 
+  const [allCollapsed, setAllCollapsed] = useState(true); 
+
+  const toggleCollapseAll = () => {
+    const newCollapsedState = {};
+    taskList.forEach((task) => {
+      const featurePropertiesList = AsMappableTask(task).osmFeatureProperties(props.osmElements);
+      featurePropertiesList.forEach((_, index) => {
+        newCollapsedState[`${task.id}-${index}`] = !allCollapsed;
+      });
+    });
+    setCollapsed(newCollapsedState);
+    setAllCollapsed(!allCollapsed); 
+  }
+
+  const toggleCollapse = (index) => {
+    setCollapsed(prevState => ({
+      ...prevState,
+      [index]: !prevState[index], 
+    }));
+  }
 
+  const propertyLists = _map(taskList, (task) => {
+    const featurePropertiesList = AsMappableTask(task).osmFeatureProperties(props.osmElements)
 
     return (
-      <QuickWidget
-        {...this.props}
-        className="task-properties-widget"
-        widgetTitle={<FormattedMessage {...messages.title} />}
-      >
-        {propertyLists}
-      </QuickWidget>
+      <div key={task.id} className="mr-mb-4 border-b mr-border-gray-300 pb-2">
+        {
+          featurePropertiesList.map((feature, index) => {
+            const featureIndex = `${task.id}-${index}`; 
+            const isCollapsed = collapsed[featureIndex]; 
+            return (
+              <div key={featureIndex} className="mr-pb-2 mr-border-b mr-bg-gray-100">
+                <div className="mr-text-yellow mr-cursor-pointer mr-p-1 mr-rounded mr-flex mr-justify-between mr-items-center" onClick={() => toggleCollapse(featureIndex)}>
+                  <div className="mr-flex mr-items-center">
+                    <div className="mr-text-yellow mr-font-bold mr-text-lg">
+                      <FormattedMessage {...messages.taskLabel} values={{taskId: task.id}}/>
+                    </div>
+                    <span className="mr-text-grey-lighter mr-mx-2 mr-mt-1">
+                      {feature.properties?.id || feature.geometry?.type}
+                    </span>
+                  </div>
+                  <div className="mr-cursor-pointer">
+                    {isCollapsed ? '▼' : '▲'}
+                  </div>
+                </div>
+                {!isCollapsed && ( 
+                  <PropertyList featureProperties={feature.properties} hideHeader
+                    lightMode={false} {...props} />
+                )}
+              </div>
```

Comment:
I think it'd be better to use a `<details>` element here; it'll be more accessible to keyboard navigation and screen reader users. The outer div would become a `<details>` and the always visible div (the one not wrapped in `isCollapsed`) would become a `<summary>` element. You can remove the `<div>{isCollapsed ? '▼' : '▲'}</div>` since you get that behavior for free with `<details>` (you can style the `::marker` pseudo-element if you want to customize it).